fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -685,6 +827,64 @@ public void updateLaoName() {
+            Log.d(TAG, LAO_FAILURE_MESSAGE);
+            return;
+        }
+        String channel = lao.getChannel();
+        try {
+            KeysetHandle publicKeysetHandle = mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+            String publicKey = Keys.getEncodedKey(publicKeysetHandle);
+            byte[] sender = Base64.getUrlDecoder().decode(publicKey);
+            PublicKeySign signer = mKeysetManager.getKeysetHandle().getPrimitive(PublicKeySign.class);
+
+            long now = Instant.now().getEpochSecond();
+            UpdateLao updateLao = new UpdateLao(publicKey, lao.getCreation(), lao.getName(), now, witnesses);
+            MessageGeneral msg = new MessageGeneral(sender, updateLao, signer, mGson);
+            Disposable disposable =
+                    mLAORepository
+                            .sendPublish(channel, msg)
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .timeout(5, TimeUnit.SECONDS)
+                            .subscribe(
+                                    answer -> {
+                                        if (answer instanceof Result) {
+                                            Log.d(TAG, "updated lao witnesses");
+                                            StateLao stateLao = new StateLao(updateLao.getId(), updateLao.getName(), lao.getCreation(), updateLao.getLastModified(), publicKey, msg.getMessageId(), updateLao.getWitnesses(), new ArrayList<>());
+                                            MessageGeneral stateMsg = new MessageGeneral(sender, stateLao, signer, mGson);
+                                            mLAORepository
+                                                    .sendPublish(channel, stateMsg)
+                                                    .subscribeOn(Schedulers.io())
+                                                    .observeOn(AndroidSchedulers.mainThread())
+                                                    .timeout(5, TimeUnit.SECONDS)
+                                                    .subscribe(
+                                                            answer2 -> {
+                                                                if (answer2 instanceof Result) {
+                                                                    Log.d(TAG, "updated lao state with new witnesses");
+                                                                } else {
+                                                                    Log.d(TAG, "failed to update lao state with new witnesses");
+                                                                }
+                                                            },
+                                                            throwable -> Log.d(TAG, "timed out waiting for result on update lao witnesses", throwable)
+                                                    );
```

Comment:
When updating the name and the witness list of a LAO a `StateLao` message is sent, creating a method for this avoids the code duplication in `updateLaoWitnesses` and `updateLaoName`